crates/compiler/builtins/bitcode/src/dec.zig

```diff
@@ -391,6 +391,29 @@ pub const RocDec = extern struct {
         }
     }
 
+    fn powInt(base: RocDec, exponent: i128) RocDec {
+        if (exponent == 0) {
+            return RocDec.one_point_zero;
+        } else if (exponent > 0) {
+            if (@mod(exponent, 2) == 0) {
+                const half_power = RocDec.powInt(base, exponent >> 1); // `>> 1` == `/ 2`
+                return RocDec.mul(half_power, half_power);
+            } else {
+                return RocDec.mul(base, RocDec.powInt(base, exponent - 1));
+            }
+        } else {
+            return RocDec.div(RocDec.one_point_zero, RocDec.powInt(base, -exponent));
+        }
+    }
+
+    fn pow(base: RocDec, exponent: RocDec) RocDec {
+        if (exponent.trunc().num == exponent.num) {
+            return base.powInt(@divTrunc(exponent.num, RocDec.one_point_zero_i128));
+        } else {
+            return fromF64(std.math.pow(f64, base.toF64(), exponent.toF64())).?;
```

Comment:
I'm not sure we can do better than this...

Reply:
Not sure if this is better or not, but we can do something like:

(base ^ wholeNumberExponent) * (base ^ fractionalExponent)

Doing only the second part as floats.

That said, probably the real answer is there is some sort of complex algorithm we need to look up an implement design specifically for fixed point math.

Oh I guess we could also do some sort of:

(base ^ wholeNumberExponent) * nthRoot(base, 1/fractionalExponent)

but for now, this is probably fine.